sympy/polys/rootoftools.py

```diff
@@ -354,7 +355,9 @@ def __new__(cls, f, x, index=None, radicals=False, expand=True):
         dom = poly.get_domain()
 
         if not dom.is_ZZ:
-            raise NotImplementedError("CRootOf is not supported over %s" % dom)
+            poly = cls._roots_Algebraic(poly)
```

Comment:
The check should be for an AlgebraicField rather than just `not dom.is_ZZ`.

Reply:
How CAN i check for AlgebraicField instead of checking for domain?

Reply:
I'm not sure of how to check for AlgebraicField here , should i check for `dom.is_QQ` ?

Reply:
The check should be that all coefficients are algebraic which could be done with Expr:
```python
In [1]: sqrt(2).is_algebraic
Out[1]: True
```
You have to be careful about `Symbol('x', algebraic=True)` though.

Reply:
So will this `all(i.is_algebraic for i in poly.all_coeffs())` would work?

Reply:
It might also be necessary to check that there are no symbols.